ray-curator/ray_curator/backends/experimental/ray_data/utils.py

```diff
@@ -0,0 +1,115 @@
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import ray
+from loguru import logger
+
+from ray_curator.backends.base import NodeInfo, WorkerMetadata
+from ray_curator.stages.base import ProcessingStage
+
+if TYPE_CHECKING:
+    import loguru
+
+
+class RayStageSpecKeys(str, Enum):
+    """String enum of different flags that define keys inside ray_stage_spec."""
+
+    IS_ACTOR_STAGE = "is_actor_stage"
+    IS_FANOUT_STAGE = "is_fanout_stage"
+
+
+def get_available_cpu_gpu_resources() -> tuple[int, int]:
+    """Get available CPU and GPU resources from Ray."""
+    available_resources = ray.available_resources()
```

Comment:
Note: Ray is sometimes slow to update value if a previous usage just ended. Not sure if it's valid for this particular use case where the func is called at when a pipeline is executed, but if the utility is used elsewhere it can sometimes print out of date result. 
I've worked around this in the past by adding a really small sleep but it's not best solution.